### DISCUSSION TITLE
Share your model training results here!! 🙌

### DISCUSSION THREAD

**Discussion by nealmick** [nealmick/bb#3] [Show and tell]
![Screen Shot 2023-02-06 at 11 46 06 AM](https://user-images.githubusercontent.com/33704976/217293348-23b43c50-fc6f-4a25-b639-32741ac364d0.png)

**Comment by nealmick** [comment]
![Screen Shot 2023-02-11 at 5 29 22 PM](https://user-images.githubusercontent.com/33704976/218283900-efabb7ce-dd18-4c79-8bb7-e65c0a232c85.png)

**Reply by nealmick** [reply]
These values are using a full season as a test hold out data set, over 1k games the model has never been trained on.  Specifically the test data validation set is the 2021 season, and the train data set is the seasons from 2020 to 2014.  The expected value is the % of games predicted correct with point spread applied.  This has been a long time in development with many long nights, and lots of coffee.  When I started building this project about 3-4 years ago I had never built a website or trained any type of model.  I barely knew python.  I have learned soo much developing this project, and am looking forward creating new features and taking this project to greater heights!

**Comment by nealmick** [comment]
![Screen Shot 2023-02-16 at 7 38 35 PM](https://user-images.githubusercontent.com/33704976/219520014-3ddc3cc8-6f86-44fb-989a-3e164eebb19d.png)

**Comment by nealmick** [comment]
![Screen Shot 2023-02-17 at 6 52 53 PM](https://user-images.githubusercontent.com/33704976/219818095-23ca1612-f535-47c5-9e48-8e2a0b12340c.png)

**Reply by nealmick** [reply]
I added projected profit to the train and eval scripts.  this is using a hold out dataset of about 250 games randomly from the last season.  Each bet is 100 $ at 110/100 odds.  Every game wins $ 190.91 and looses 100 $ .  If the predicted score deviates more than 1 point compared to the point spread, it is considered a 1 point margin game, and 2,3 so on.  in the screen shot the 3 points margins got 25/40 62.5%, spending 4000$ and profiting $ 773 .   Out of the 250 test games 40 games had more than a 3 point difference between the spread and prediction.  Out of those 40 games 25 of them won against the spread. 40 games at 100 $ each = 4k and 25 games won at 190$ = 4750$ profit = 750 $.  Using less than 250 games ive found the results unreliable (too much variance). but there are 1200 games in a season.  Maybe im just wrong here but as far as i can tell this process could be repeated 4-5 times a season 1200/250, each batch of 250 netting up to 10% profit.  Continuing along this logic that means you would end up with upwards of 40% profit per season.  absolutely incredible!

**Comment by nealmick** [comment]
Take's some time to get a model that converges properly, without overfitting.  Results should hold up to future games,  they are calculated using a holdout dataset (test games).  Overfitting is when the model effectively remembers previous game results rather than learning the essence of the game compared to the inputs, causing poor performance on new games never seen before, and good performance on games the model remembers and has seen before.  This happens when the model docent have enough unique data features that correlate to the output sufficiently.  Improving results could be accomplished by using a bigger dataset (more games), or using more input features per game.  Currently im pretty happy with the results, ive been able to get models trained up to 60-70% accuracy on the margin 3 and 4 metrics using 750+ test games the model has never seen.

![Screen Shot 2023-03-03 at 2 28 39 PM](https://user-images.githubusercontent.com/33704976/222812024-ea1e1aaa-618b-4e36-b42e-bc27f3242a19.png)

**Comment by nealmick** [comment]
Did some testing today got this model:
<img width="525" alt="Screen Shot 2023-03-16 at 4 54 08 PM" src="https://user-images.githubusercontent.com/33704976/225750633-252e5a82-9b6a-4e2f-b3dc-8c2e0ac716b0.png">

export all games from the 1st of this month:
https://docs.google.com/spreadsheets/d/1g9luwBxhRdUS076_snFhHAJXQjafmbqRbKEAjjl695U/edit?usp=sharing

61% on margin 3 from the past 2 weeks, pretty good! made this model the default now.
